src/driver.h

```diff
@@ -81,10 +81,10 @@ class DriverFactoryBase {
 	 * @param type The type to get the driver for.
 	 * @return The active driver.
 	 */
-	static Driver **GetActiveDriver(Driver::Type type)
+	static std::unique_ptr<Driver> &GetActiveDriver(Driver::Type type)
 	{
-		static Driver *s_driver[3] = { nullptr, nullptr, nullptr };
-		return &s_driver[type];
+		static std::unique_ptr<Driver> s_driver[3] = { nullptr, nullptr, nullptr };
```

Comment:
Maybe something along the lines of the following? So, when we add a new driver for whatever reason the number of unique pointers gets updated automatically. Other than this, it looks fine to me.
```suggestion
		static std::array<std::unique_ptr<Driver>, DT_END> s_driver = {};
```